pkg/controller/awsprivatelink/awsprivatelink_controller_test.go

```diff
@@ -783,6 +743,12 @@ users:
 			m.EXPECT().DescribeLoadBalancers(gomock.Any()).
 				Return(nil, awserr.New("LoadBalancerNotFound", "Loadbalance could not be found", nil))
 		},
+		expectedConditions: []hivev1.ClusterDeploymentCondition{{
+			Status:  corev1.ConditionFalse,
+			Type:    hivev1.AWSPrivateLinkReadyClusterDeploymentCondition,
+			Reason:  "DiscoveringNLBNotYetFound",
+			Message: "discovering NLB for the cluster, but it does not exists yet",
+		}},
```

Comment:
`getExpectedConditions` is not used here. It will be helpful if you can document when to use this function.